core/src/main/java/jenkins/model/Jenkins.java

```diff
@@ -4976,14 +4993,49 @@ private static void computeVersion(ServletContext context) {
      * Version number of this Jenkins.
      */
     public static String VERSION = UNCOMPUTED_VERSION;
-
+    
+    /**
+     * Minimal Java level required to run this Jenkins.
+     * If {@code null}, the version is unknown.
+     */
+    private static @CheckForNull String JAVA_MIN_LEVEL = null;
+    private static VersionNumber DEFAULT_JAVA_MIN_LEVEL = new VersionNumber("8");
```

Comment:
> This change just adds new API which can be used to determine the current requirement

But if that is not met how can you get it (the class format will not be compatable)?   if you can get it - you don't care as the requirement must have been met...